### PR TITLE
feat: add env configuration + jsdom custom conf

### PR DESCRIPTION
**Pull request by userquin**
This PR adds `environmentConfiguration` and `jsdom` configuration.
I need some help for `happy-dom`.

### CODE REVIEW

**Comment by dammy001** [comment]
@userquin is it just to add happy-dom file and config?

**Comment by userquin** [comment]
@dammy001 I mean if this makes sense in `happy-dom` (it is just an example)?

imagen

File: packages/vitest/src/types/config.ts [hunk 1]
```diff
@@ -73,6 +73,11 @@ export interface InlineConfig {
    */
   environment?: BuiltinEnvironment
 
+  /**
+   * The configuration for the environment.
+   */
+  environmentConfiguration?: Record<string, any>
```
Comment (antfu) [inline packages/vitest/src/types/config.ts:79]:
```suggestion
  environmentOptions?: { jsdom?: JSdomOptions, ... }
```

Comment (userquin) [reply packages/vitest/src/types/config.ts]: we need to check `resources` entry on `JSDOMOptions` since I don't expose the `jsdom` types.

**Pull request merged**